components/provisioner/internal/provisioning/service.go

```diff
@@ -156,10 +162,60 @@ func (r *service) DeprovisionRuntime(id, tenant string) (string, apperrors.AppEr
 	return operation.ID, nil
 }
 
+func (r *service) UpgradeGardenerShoot(runtimeID string, input gqlschema.UpgradeShootInput) (*gqlschema.OperationStatus, apperrors.AppError) {
+	log.Infof("Starting Upgrade of Gardener Shoot for Runtime '%s'...", runtimeID)
+
+	if input.GardenerConfig == nil {
+		return &gqlschema.OperationStatus{}, apperrors.Internal("error: Gardener config is nil")
+	}
+
+	session := r.dbSessionFactory.NewReadWriteSession()
+
+	err := r.verifyLastOperationFinished(session, runtimeID)
+	if err != nil {
+		return &gqlschema.OperationStatus{}, err
+	}
+
+	cluster, dberr := session.GetCluster(runtimeID)
+	if dberr != nil {
+		return &gqlschema.OperationStatus{}, apperrors.Internal("failed to find shoot cluster to upgrade: %s", dberr.Error())
```

Comment:
This error message might be confusing indicating that we failed to find Shoot CR not a Cluster in the DB.
I also think we should stick to one convention with capitalizing errors as most of them in this file start with the upper case.